extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/model/Range.java

```diff
@@ -62,10 +62,10 @@ public long getDuration() {
      * Returns true, if the given calendar matches into the range.
      */
     public boolean matches(Calendar cal) {
-        if (start != null && end != null) {
-            return cal.getTimeInMillis() >= start.getTimeInMillis() && cal.getTimeInMillis() < end.getTimeInMillis();
-        }
-        return false;
+        long matchStart = start != null ? start.getTimeInMillis()
+                : DateTimeUtils.truncateToMidnight(cal).getTimeInMillis();
+        long matchEnd = end != null ? end.getTimeInMillis() : DateTimeUtils.truncateToMidnight(cal).getTimeInMillis();
+        return cal.getTimeInMillis() >= matchStart && cal.getTimeInMillis() < matchEnd;
```

Comment:
If no start/end are given, then both conditions are comparing against `DateTimeUtils.truncateToMidnight(cal)`. How can this ever be true? Doesn't `matchEnd` needs to add 24h?

Reply:
Range start/end are always for the current day. If no start/end are given (e.g night in northern hemisphere), there is nothing to match and should return false. So it's OK.

Reply:
oh, I see. Thanks for the clarification!

Reply:
No, wait, I still don't fully get it - can it happen that only one is given? I.e. if there is only a start, then it would also return false (because the second part is using the previous midnight), even if cal is after matchStart.

Reply:
you are right! end must be truncated to endOfDay and not to midnight. Thanks!!!